src/proxychains-cli

```diff
@@ -0,0 +1,192 @@
+  --proxychains-path PATH
+                          Path to proxychains4 (default: proxychains4 in PATH)
+
+Other:
+  -h, --help              Show this help
+  -v, --version           Show version
+
+Examples:
+  proxychains-cli -P "socks5 127.0.0.1 1080" curl example.com
+  proxychains-cli -P "socks5 127.0.0.1 1080" curl -v example.com
+  proxychains-cli -P "socks5 127.0.0.1 1080 user pass" wget file.tar.gz
+EOF
+}
+
+show_version() {
+  echo "proxychains-cli ${PROXYCHAINS_CLI_VERSION}"
+}
+
+# Parse arguments
+while [ $# -gt 0 ]; do
+case "$1" in
+-c|--chain)
+  chain_mode="$2"
+  shift 2
+  ;;
+-l|--chain-len)
+  chain_len="$2"
+  shift 2
+  ;;
+-q|--quiet)
+  quiet=1
+  shift
+  ;;
+-d|--dns)
+  dns_mode="$2"
+  shift 2
+  ;;
+-S|--dns-subnet)
+  dns_subnet="$2"
+  shift 2
+  ;;
+-R|--read-timeout)
+  read_timeout="$2"
+  shift 2
+  ;;
+-T|--connect-timeout)
+  connect_timeout="$2"
+  shift 2
+  ;;
+-n|--localnet)
+  localnets="${localnets}${2}
+"
+  shift 2
+  ;;
+--dnat)
+  # Parse dnat string: "SRC DST" or "SRC:PORT DST:PORT"
+  dnats="${dnats}${2}
+"
+  shift 2
+  ;;
+-P|--proxy)
+  # Parse proxy string: "TYPE IP PORT [USER PASS]"
+  proxy_str="$2"
+  # Convert spaces to tabs for .conf format
+  entry=$(echo "$proxy_str" | sed 's/  */ /g' | tr ' ' '\t')
+  proxies="${proxies}${entry}
+"
+  shift 2
+  ;;
+-h|--help)
+  show_help
+  exit 0
+  ;;
+-v|--version)
+  show_version
+  exit 0
+  ;;
+--config)
+  show_config=1
+  shift
+  ;;
+--proxychains-path)
+  proxychains_path="$2"
+  shift 2
+  ;;
+-*)
+  echo "Unknown option: $1" >&2
+  echo "Use -h for help" >&2
+  exit 1
+  ;;
+*)
+  # First non-option is program, stop parsing
+  break
+  ;;
+esac
+done
+
+# Check if program specified (unless --config)
+if [ $# -eq 0 ] && [ -z "$show_config" ]; then
+  echo "Error: No program specified" >&2
+  echo "Use -h for help" >&2
+  exit 1
+fi
+
+# Check if at least one proxy specified
+if [ -z "$proxies" ]; then
+  echo "Error: No proxy specified. Use -P to add proxy." >&2
+  exit 1
+fi
+
+# Build DNS config line
+dns_config=""
+case "$dns_mode" in
+  proxy) dns_config="proxy_dns";;
+  old) dns_config="proxy_dns_old";;
+  off|none|"") ;;
+  *) dns_config="proxy_dns_daemon $dns_mode";;
+esac
+
+# Generate config content
+config_content="${chain_mode}_chain
+${chain_len:+chain_len = $chain_len}
+${quiet:+quiet_mode}
+${dns_config}
+${dns_subnet:+remote_dns_subnet $dns_subnet}
+${read_timeout:+tcp_read_time_out $read_timeout}
+${connect_timeout:+tcp_connect_time_out $connect_timeout}
+$(printf '%s' "$localnets" | while IFS= read -r net; do [ -n "$net" ] && echo "localnet $net"; done)
+$(printf '%s' "$dnats" | while IFS= read -r m; do [ -n "$m" ] && echo "dnat $m"; done)
+[ProxyList]
+$(printf '%s' "$proxies")"
+
+# If --config, print and exit
+if [ -n "$show_config" ]; then
+  printf '%s\n' "$config_content"
+  exit 0
+fi
+
+# Execute with config via environment variable
+export PROXYCHAINS_CONFIG_DATA="$config_content"
```

Comment:
ah, now you opted to pass the entire config as an environment variable instead of a pipe. why's that ?